sw/device/lib/dif/dif_kmac.h

```diff
@@ -375,14 +382,74 @@ typedef enum dif_kmac_error {
  * The state of the message FIFO used to buffer absorbed data.
  */
 typedef enum dif_kmac_fifo_state {
-  /** The message FIFO is empty. */
-  kDifKmacFifoStateEmpty,
   /** The message FIFO is not empty or full. */
-  kDifKmacFifoStatePartial,
+  kDifKmacFifoStatePartial = 0,
```

Comment:
Is there a reason this one should be 0 while the other two have a one-hot encoding? It seems to me like it would make sense to have `1 << 0`, `1 << 1`, and `1 << 2` for the three options here.

Reply:
The reason is explained [here](https://github.com/lowRISC/opentitan/pull/12125/commits/f6462aa229e17a8e4319021500bdb9cba63baf69#r852835026)
Basically when the two bits are zero, it means that the fifo is partially filled.

Reply:
That link doesn't work for me and I can't find an explanation like that anywhere near this code; would you mind adding whatever the explanation is to a comment here?

Reply:
done.